File: 0075-sort-colors/0075-sort-colors.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    void sortColors(vector<int>& nums) {
    int l = 0, m = 0, h = nums.size() - 1;
    while (m <= h) {
        if (nums[m] == 0) {
            swap(nums[l], nums[m]);
            l++;
            m++;
        } else if (nums[m] == 1) {
            m++;
        } else {
            swap(nums[m], nums[h]);
            h--;
        }
    }
}

int main() {
    vector<int> n = {2, 0, 2, 1, 1, 0};

    cout << "Original Array:" << endl;
    for (int num : n) {
        cout << num << " ";
    }
    cout << endl;

    sortColors(n);

    cout << "Sorted Array:" << endl;
    for (int num : n) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n=nums.length;
        for(int i=n-1;i>=0;i--){
            for(int j=0;j<i;j++){
                if(nums[j]>nums[j+1]){
                    int temp =nums[j];
                    nums[j]=nums[j+1];
                    nums[j+1]=temp;
                }
            }
        }
    }
}